generate_homepage_from_content.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
+
+import argparse
+import csv
+import json
+import os.path
+import requests
+import sys
+import textwrap
+import urllib
+
+# default filepaths
+DEFAULT_CONTENT_CSV_PATH = './content.csv'
+DEFAULT_PAGE_TEMPLATE_PATH = './index_template.html'
+DEFAULT_OUTFILE_PATH = 'index.html'
+
+# CSV keys and accepted values
+CSV_KEY_TYPE = 'type'
+CSV_KEY_URL = 'url'
+CSV_KEY_QUOTE = 'quote'
+CONTENT_TYPE_TWITTER = 'twitter'
+CONTENT_TYPE_INSTAGRAM = 'instagram'
+CONTENT_TYPE_WEBSITE = 'website'
+CONTENT_TYPES = [CONTENT_TYPE_TWITTER, CONTENT_TYPE_INSTAGRAM, CONTENT_TYPE_WEBSITE]
+EMBEDDED_CONTENT_TYPES = [CONTENT_TYPE_TWITTER, CONTENT_TYPE_INSTAGRAM]
+
+# HTML templates
+CONTENT_CONTAINER = '<div class="content %s">%s</div>'
+WEBSITE_CONTENT_TEMPLATE = ('<span class="website-content">%s</span>\n'
+                            '<span class="website-url">%s</span>')
+
+# API endpoints and keys
+TWITTER_OEMBED_ENDPOINT = 'https://publish.twitter.com/oembed?url=%s'
+TWITTER_HTML_KEY = 'html'
+INSTAGRAM_OEMBED_ENDPOINT = 'https://api.instagram.com/oembed/?url=%s'
+INSTAGRAM_HTML_KEY = 'html'
+
+
+def parse_command_line_args():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=textwrap.dedent("""\
+        Generate an index.html file for www.notmylockerroom.com that formats
+        the content provided (as a csv) into html (hitting the appropriate
+        API to get embed code where necessary). The template html file should
+        contain a '%s' where content should be inserted.
+
+        Each row of the content csv represents a content element that will be
+        put into the page. Columns should be 'type', 'url', and 'quote'.
+
+        Type: one of 'twitter', 'instagram', or 'website'.
+        Url: the url of the content (if twitter/instagram, a direct link to the
+            tweet/photo)
+        Quote: for 'website' only, the quote to highlight.
+    """))
+    parser.add_argument('--content_csv', type=str,
+                        help=('path to csv containing content (default: %s)' %
+                              DEFAULT_CONTENT_CSV_PATH))
+    parser.add_argument('--page_template', type=str,
+                        help=('path to page template into which to insert '
+                              'content (default: %s)' %
+                              DEFAULT_PAGE_TEMPLATE_PATH))
+    parser.add_argument('--outfile', type=str,
+                        help=('path to write completed file to (default: %s)' %
+                              DEFAULT_OUTFILE_PATH))
+    return parser.parse_args()
+
+
+def get_twitter_embed_code(url):
```

Comment:
Seems like you could combine get_twitter_embed_code and get_instagram_embed_code into one data-driven function,and even control the whole shebang with a dict keyed by content_type, with fields for content_container, oembed_endpoint, etc. Then adding a new input type would just mean adding a new entry to the dict.